logic/src/commonMain/kotlin/com/wire/kalium/logic/network/SessionManagerImpl.kt

```diff
@@ -31,14 +33,17 @@ class SessionManagerImpl internal constructor(
     private val idMapper: IdMapper = MapperProvider.idMapper()
 ) : SessionManager {
 
-    private val session: Atomic<SessionDTO?> = Atomic(null)
+    private val lock: Mutex = Mutex()
+    private var session: SessionDTO? = null
     private var serverConfig: Atomic<ServerConfigDTO?> = Atomic(null)
 
-    override fun session(): SessionDTO = session.get() ?: run {
-        wrapStorageRequest { tokenStorage.getToken(idMapper.toDaoModel(userId)) }
-            .map { sessionMapper.fromEntityToSessionDTO(it) }
-            .onSuccess { session.set(it) }
-        session.get()!!
+    override suspend fun session(): SessionDTO = lock.withLock {
```

Comment:
So I take it that withLock will do the following:

lock
execute lambda
unlock